packages/aws-cdk/lib/api/cxapp/exec.ts

```diff
@@ -42,38 +40,42 @@ export async function execProgram(aws: SDK, config: Configuration): Promise<cxap
     context[cxapi.DISABLE_VERSION_REPORTING] = true;
   }
 
-  const stagingDir = config.settings.get(['staging']);
-  context[cxapi.ASSET_STAGING_DIR_CONTEXT] = stagingDir;
+  let stagingEnabled = config.settings.get(['staging']);
+  if (stagingEnabled === undefined) {
+    stagingEnabled = true;
+  }
+  if (!stagingEnabled) {
+    context[cxapi.DISABLE_ASSET_STAGING_CONTEXT] = true;
+  }
 
   debug('context:', context);
-
   env[cxapi.CONTEXT_ENV] = JSON.stringify(context);
 
   const app = config.settings.get(['app']);
   if (!app) {
     throw new Error(`--app is required either in command-line, in ${PROJECT_CONFIG} or in ${USER_DEFAULTS}`);
   }
 
+  // by pass "synth" if app points to a cloud assembly
+  if (await fs.pathExists(app) && (await fs.stat(app)).isDirectory()) {
+    debug('--app points to a cloud assembly, so we by pass synth');
+    return new cxapi.CloudAssembly(app);
+  }
+
   const commandLine = await guessExecutable(appToArray(app));
 
-  const outdir = await fs.mkdtemp(path.join(os.tmpdir(), 'cdk'));
+  const outdir = config.settings.get([ 'output' ]);
+  if (!outdir) {
+    throw new Error('--output must be specified');
```

Comment:
Oh no. Where will this value come from?

Reply:
output is always set, so that's unlikely to happen unless it's a bug. Maybe I'll change the message.

Reply:
changed to: `unexpected: --output is required`